src/views/publish/cube-preview.ejs

```diff
@@ -0,0 +1,291 @@
+<%- include("../partials/header", t); %>
+
+<div class="govuk-width-container app-width-container">
+    <main class="govuk-main-wrapper" id="main-content" role="main">
+
+        <%- include("../partials/error-handler"); %>
+
+        <div class="govuk-panel govuk-panel--confirmation">
+            <p class="govuk-panel__title-m">
+                <%= t('publish.cube_preview.panel') %>
+            </p>
+        </div>
+
+        <h2 class="govuk-heading-l">
+            <%= t('dataset_view.key_information') %>
+        </h2>
+
+        <%
+            const publishDate = new Date(Math.max.apply(null, dataset.revisions.map((rev) => new Date(Date.parse(rev.publish_at? rev.publish_at : rev.created_at)))));
+            let updateObj = { years: 0 };
+            if (dataset.datasetInfo.update_frequency) {
+                if (dataset.datasetInfo.update_frequency.frequency_unit === 'year')
+                    updateObj = { years: dataset.datasetInfo.update_frequency.frequency_value };
+                if (dataset.datasetInfo.update_frequency.frequency_unit === 'month')
+                    updateObj = { months: dataset.datasetInfo.update_frequency.frequency_value };
+                if (dataset.datasetInfo.update_frequency.frequency_unit === 'week')
+                    updateObj = { weeks: dataset.datasetInfo.update_frequency.frequency_value };
+                if (dataset.datasetInfo.update_frequency.frequency_unit === 'day')
+                    updateObj = { days: dataset.datasetInfo.update_frequency.frequency_value };
+            }
```

Comment:
Same here - don't do calculations in the template, just pass `nextUpdate` through from the route handler.

It can probably just be 

```ts
const update = dataset.datasetInfo.update_frequency;
const nextUpdate = update ? dateAdd(publishDate, { [update.frequency_unit]: update.frequency_value }) : undefined;
```